### PR TITLE
refactor inflation features

### PR DESCRIPTION
**Pull request by yoosah**
- get total supply from genesis state
- pre-mint inflation amount
- unify reward purse
- apply current block height to step
- modify annual inflation ratio to 4.88%
- remove step_claim

### CODE REVIEW

File: execution-engine/contracts/hdac-system/client-api-proxy/src/client_api/mod.rs [hunk 1]
```diff
@@ -46,7 +46,7 @@ pub enum Api {
     Unbond(Option<U512>),
     StandardPayment(U512),
     TransferToAccount(PublicKey, U512),
-    Step(),
+    Step(u64),
```
Comment (jhyeom26) [inline execution-engine/contracts/hdac-system/client-api-proxy/src/client_api/mod.rs]: `step` is not available for general users. It seems correct to delete this API in client_api_proxy.

File: execution-engine/contracts/hdac-system/pop/src/pop_contract/mod.rs [hunk 2]
```diff
@@ -129,35 +132,40 @@ impl ProofOfProfessionContract {
...
-
-        // Actual mint & transfer will be done at client-proxy
+        if let Ok(premint_purse) = get_purse(uref_names::POS_PREMINT_PURSE) {
+            let commission_amount = store::read_commission_amount(validator);
+            let transfer_res: TransferResult =
+                system::transfer_from_purse_to_account(
+                    premint_purse, *validator, commission_amount);
+            if let Err(err) = transfer_res {
+                runtime::revert(err);
+            }
```
Comment (jhyeom26) [inline execution-engine/contracts/hdac-system/pop/src/pop_contract/mod.rs]: Hope to change the way to deal with the error.
```rust
system::transfer_from_purse_to_purse(payment_purse, communtiy_purse, remain_amount)	
                .map_err(|_| Error::FailedTransferToPremintPurse)?;
```

runtime::revert invalidates all behavior of that invoking. so this makes it to be careful to call `claim_commission`.

File: execution-engine/contracts/hdac-system/pop/src/pop_contract/mod.rs [hunk 3]
```diff
@@ -129,35 +132,40 @@ impl ProofOfProfessionContract {
...
-        // Actual mint & transfer will be done at client-proxy
+        // Processing commission claim table
+        if let Ok(premint_purse) = get_purse(uref_names::POS_PREMINT_PURSE) {
+            let reward_amount = store::read_reward_amount(user);
+            let transfer_res: TransferResult =
+                system::transfer_from_purse_to_account(
+                    premint_purse, *user, reward_amount);
+            if let Err(err) = transfer_res {
+                runtime::revert(err);
+            }
```
Comment (jhyeom26) [inline execution-engine/contracts/hdac-system/pop/src/pop_contract/mod.rs]: same here

File: execution-engine/contracts/hdac-system/pop-install/src/lib.rs [hunk 4]
```diff
@@ -69,24 +79,24 @@ pub extern "C" fn call() {
...
+    premint_amount: U512,
 ) -> BTreeMap<String, Key> {
     let mint = ContractRef::URef(URef::new(mint_uref.addr(), AccessRights::READ));
     let mut named_keys = BTreeMap::<String, Key>::default();
 
-    let total_bonds = genesis_validators.values().fold(U512::zero(), |x, y| x + y);
     let bonding_purse = mint_purse(&mint, total_bonds);
     let payment_purse = mint_purse(&mint, U512::zero());
+    let premint_purse = mint_purse(&mint, premint_amount);
     let rewards_purse = mint_purse(&mint, U512::zero());
```
Comment (jhyeom26) [inline execution-engine/contracts/hdac-system/pop-install/src/lib.rs:91]: where the `available_amount` is minted?

File: execution-engine/contracts/hdac-system/pop/src/pop_contract/mod.rs [hunk 5]
```diff
@@ -129,35 +132,40 @@ impl ProofOfProfessionContract {
...
+        if let Ok(premint_purse) = get_purse(uref_names::POS_PREMINT_PURSE) {
+            let commission_amount = store::read_commission_amount(validator);
+            let transfer_res: TransferResult =
+                system::transfer_from_purse_to_account(
+                    premint_purse, *validator, commission_amount);
+            if let Err(err) = transfer_res {
+                runtime::revert(err);
+            }
+            store::write_commission_amount(validator, U512::zero());
+        }
```
Comment (jhyeom26) [inline execution-engine/contracts/hdac-system/pop/src/pop_contract/mod.rs]: Purse lookup error handling is omitted.

File: execution-engine/contracts/hdac-system/pop/src/pop_contract/mod.rs [hunk 6]
```diff
@@ -129,35 +132,40 @@ impl ProofOfProfessionContract {
...
+        if let Ok(premint_purse) = get_purse(uref_names::POS_PREMINT_PURSE) {
+            let reward_amount = store::read_reward_amount(user);
+            let transfer_res: TransferResult =
+                system::transfer_from_purse_to_account(
+                    premint_purse, *user, reward_amount);
+            if let Err(err) = transfer_res {
+                runtime::revert(err);
+            }
+            store::write_reward_amount(user, U512::zero());
+        }
```
Comment (jhyeom26) [inline execution-engine/contracts/hdac-system/pop/src/pop_contract/mod.rs]: Purse lookup error handling is omitted.

**Review by nowooj — APPROVED** [review]

**Review by psy2848048 — APPROVED** [review]

**Pull request merged**